File: framework/src/main/java/org/example/core/page/presentation/AbstractIPage.java
package org.example.core.page.presentation;

import org.example.core.dispatcher.IIntentDispatcher;

import javax.swing.*;

public abstract class AbstractIPage {
    private final String title;
    private final AbstractController controller;

    public AbstractIPage(String title, AbstractController controller) {
        this.title = title;
        this.controller = controller;
    }

    public final String getTitle() {
        return title;
    }

    public JPanel getPanel() {
        return controller.getLayout();
    }

    public final void setIntentDispatcher(IIntentDispatcher dispatcher) {
        controller.setIntentDispatcher(dispatcher);
    }
}

File: framework/src/main/kotlin/org/example/core/page/presentation/AbstractIPage.kt
package org.example.core.page.presentation

import org.example.core.dispatcher.IIntentDispatcher
import javax.swing.JPanel

abstract class AbstractIPage(val title: String?, private val controller: AbstractController?) {
    val panel: JPanel?
        get() = controller?.layout

    fun setIntentDispatcher(dispatcher: IIntentDispatcher?) {
        controller?.setIntentDispatcher(dispatcher)
    }
}